site/profile/manifests/route53.pp

```diff
@@ -0,0 +1,13 @@
+class profile::route53 (
+	String $domain		='graduway.com.',
+	String $environment	= undef,
+	String $hostname	= undef,
+	String $ip_address	= undef,
```

Comment:
Note that you could place the variables set via hiera as defaults here.
Result would be the same, without the need of specifying hiera data (other than the class inclusion under psick::base::*_classes.
Anyway the final result would be the same.

Also note that if you have undef as default value, and data type as String (and not Optional[String]) and you don't explicitly configure on hiera these params, you will get a wrong type error.
(Which is fine, in this case, as you don't want to create route53 records with missing data)

Reply:
Is it better for me to specify the default from hiera common instead of using the default configuration that I did?

Reply:
@shapi78 Generally speaking I'd recomment to specify defaults as in the class params whenever the default value is sane, safe and usable without issues.
If a specific value, which can't have a sane default, has to be set for a parameter, then I'd leave the parameter without default value, so that you need to specify it via hiera.
In any way the final result is the same, if all the needed params value are passed either via hiera or are present as defaults.